x/amm/keeper/msg_server_swap_exact_amount_in.go

```diff
@@ -17,9 +17,13 @@ func (k msgServer) SwapExactAmountIn(goCtx context.Context, msg *types.MsgSwapEx
 		return nil, err
 	}
 
+	recipient, err := sdk.AccAddressFromBech32(msg.Recipient)
+	if err != nil {
+		recipient = sender
```

Comment:
@jelysn I would suggest to throw an error that says invalid recipient address and stop the flow if that happens so the user expecting to receive the fund in the receipt address doesn't get confused as it gets send to the original sender address what do you think?

Reply:
This is the case recipient is empty - when recipient is empty it's returning to sender address, here msg.Recipient is an optional field - for the users using amm directly won't configure recipient but only the trade shield contract will use custom recipient.